kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -275,55 +275,102 @@ class LineSearchStrategy
         typename TBuilderAndSolverType::Pointer pBuilderAndSolver = this->GetBuilderAndSolver();
 
         TSystemVectorType aux(b.size()); //TODO: do it by using the space
-        TSparseSpace::Assign(aux,0.5, Dx);
-
-        //compute residual without update
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double ro = TSparseSpace::TwoNorm(b);
-
-        //compute half step residual
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rh = TSparseSpace::TwoNorm(b);
-
-        //compute full step residual (add another half Dx to the previous half)
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
+        
+        double x1 = 0.5;
```

Comment:
I would like to know where this values are coming

Reply:
I don't like hardcoded values

Reply:
I see your point. I will change this, once it is decided that it is actually worth the change.

Reply:
let me clarify: when we decide that we want to keep this implementation of line_search (which is a change wrt previous behaviour)

Reply:
You can use the Parameters, presenving as default the current values. But this implementation requires to hardmodify in case you want, more iterations, different X1, etc...

Reply:
The change is not that much

Reply:
yes, as i am telling i am ok with making the change. what i am not sure is if we want this implementaion or if we prefer the previous one.